codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/waiters/KotlinJmespathExpressionVisitor.kt

```diff
@@ -244,20 +246,41 @@ class KotlinJmespathExpressionVisitor(
     }
 
     override fun visitMultiSelectHash(expression: MultiSelectHashExpression): VisitedExpression {
-        throw CodegenException("MultiSelectHashExpression is unsupported")
+        val properties = mutableListOf<String>()
+        expression.expressions.forEach {
+            properties.add("val ${it.key}: T")
+        }
+        writer.write("class Selection<T>(${properties.joinToString()})")
```

Comment:
**Nit**: Mutable lists are rarely necessary within a block. This, for instance, can be written as a single `joinToString` translation:

```kotlin
val properties = expression.expressions.joinToString { "val ${it.key}: T" }
writer.write("class Selection<T>($properties)")
```